File: Java/프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.java
class Solution {
    public int[] solution(int num, int total) {
        int[] answer = new int[num];
        int result=0;
        int n=-100;
        int temp=0;
        while(true){
            temp=n;
            for(int i=0; i<num;i++){
                result+=n;
                n++;
            }
            n--;
            if(result==total){
                for(int i=0; i<answer.length ; i++){
                    answer[answer.length-1-i]=n;
                    n--;
                }
                break;
            }
            else {
                result=0;
                n=temp+1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.js
function solution(num, total) {
    var answer = [];
    answer = new Array(num);
    let result=0;
    let n=-100;
    let temp=0;
    while(true){
        temp=n;
        for(let i=0; i<num;i++){
            result+=n;
            n++;
        }
        n--;
        if(result==total){
            for(let i=0; i<answer.length ; i++){
                answer[answer.length-1-i]=n;
                n--;
            }
            break;
        }
        else {
            result=0;
            n=temp+1;
        }
    }
    return answer;
}